apps/podcast-transcribe-episode/src/python/podcast_transcribe_episode/workflow_interface.py

```diff
@@ -0,0 +1,302 @@
+        # configuration ones too because on programming / configuration bugs we can just fix up some code or
+        # configuration, deploy the fixes and let the workflow system automagically continue on with the workflow
+        # without us having to dig out what exactly has failed and restart things.
+        #
+        # However, on "permanent" errors (the ones when some action decides that it just can't proceed with this
+        # particular input, e.g. process a story that does not exist) there's no point in retrying anything.
+        # anything anymore.
+        McPermanentError.__name__,
+
+    ],
+
+)
+"""
+Retry parameters.
+
+https://docs.temporal.io/docs/concept-activities/
+"""
+
+
+class AbstractPodcastTranscribeActivities(object):
+    """Activities interface."""
+
+    @classmethod
+    def _create_config(cls) -> PodcastTranscribeEpisodeConfig:
+        """
+        Create and return configuration instance to be used for running the workflow.
+
+        Might get overridden in case some configuration changes have to be made while running the tests.
+        """
+        return PodcastTranscribeEpisodeConfig()
+
+    def __init__(self):
+        super().__init__()
+        self.config = self._create_config()
+
+    @activity_method(
+        task_queue=TASK_QUEUE,
+
+        # ScheduleToStart is the maximum time from a Workflow requesting Activity execution to a worker starting its
+        # execution. The usual reason for this timeout to fire is all workers being down or not being able to keep up
+        # with the request rate. We recommend setting this timeout to the maximum time a Workflow is willing to wait for
+        # an Activity execution in the presence of all possible worker outages.
+        # schedule_to_start_timeout=None,
+
+        # StartToClose is the maximum time an Activity can execute after it was picked by a worker.
+        start_to_close_timeout=timedelta(seconds=60),
+
+        # ScheduleToClose is the maximum time from the Workflow requesting an Activity execution to its completion.
+        # schedule_to_close_timeout=None,
+
+        # Heartbeat is the maximum time between heartbeat requests. See Long Running Activities.
+        # (https://docs.temporal.io/docs/concept-activities/#long-running-activities)
+        # heartbeat_timeout=None,
+
+        retry_parameters=DEFAULT_RETRY_PARAMETERS,
+    )
+    async def identify_story_bcp47_language_code(self, stories_id: int) -> Optional[str]:
+        """
+        Guess BCP 47 language code of a story.
+
+        https://cloud.google.com/speech-to-text/docs/languages
+
+        :param stories_id: Story to guess the language code for.
+        :return: BCP 47 language code (e.g. 'en-US') or None if the language code could not be determined.
+        """
+        raise NotImplementedError
+
+    @activity_method(
+        task_queue=TASK_QUEUE,
+        # schedule_to_start_timeout=None,
```

Comment:
Guessing you did mean to comment this line out—what was your reason for including this and the other commented-out parameters in this file?

Reply:
I did, this again was supposed to be a reference of what arguments (various flavors of timeouts) are available to each workflow / activity but maybe I should move them to docs or something.